### PR TITLE
Add support for `ConsensusParametersVersion::V2`

### PR DESCRIPTION
**Pull request by rafal-ch**
Partial implementation of https://github.com/FuelLabs/fuel-core/issues/2133.

## Linked Issues/PRs
(https://github.com/FuelLabs/fuel-vm/pull/821)

## Description
It adds handling for the updated data struct `ConsensusParameters` (new field: `block_transaction_size_limit`). No additional logic around this new parameter is implemented in this PR, it's just a stub for further work.

The new parameter is exposed via GraphQL, example:

Request snippet:
```graphql
  consensusParameters(version: 0) {
    blockGasLimit
    blockTransactionSizeLimit
  }
```

Response snippet:
```json
  "consensusParameters": {
    "blockGasLimit": "30000000",
    "blockTransactionSizeLimit": "129024"
  }
```

## Checklist
- [ ] Breaking changes are clearly marked as such in the PR description and changelog
- [ ] New behavior is reflected in tests
- [ ] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [ ] I have reviewed the code myself
- [ ] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

### CODE REVIEW

File: crates/fuel-core/src/schema/tx/input.rs
Comment (acerone85) [inline crates/fuel-core/src/schema/tx/input.rs]: Taking a guess here, but I think it's because converting to hex format ensures that all characters are in utf-8, which is an  assumption of most json decoders.

**Comment by acerone85** [comment]
When I run the node and make the following query: 
```
curl -X POST http://localhost:4000/v1/graphql \
-H "Content-Type: application/json" \
-d '{
    "query": "{ consensusParameters(version:0) { version blockGasLimit blockTransactionSizeLimit } }"
}'
```

I get the following result
```json
{
  "data": {
    "consensusParameters": {
      "version": "V2",
      "blockGasLimit": "30000000",
      "blockTransactionSizeLimit": "129024"
    }
  }
}
```

Is this the expected result? I would imagine that by querying `consensusParameters(version:0)` we got still the V1 parameters, and to get the V2 consensus parameters we would need to run a query with `consensusParameters(version:1)`.

Or am I missing something?

File: crates/fuel-core/src/schema/tx/input.rs
Comment (rafal-ch) [reply crates/fuel-core/src/schema/tx/input.rs]: My conjecture is that [this recent change](https://github.com/FuelLabs/fuel-vm/pull/670/files#diff-5a170feaca83c54fdc3cde01fd6b6bdfe23d0f7c983e5f1bf8d6fd5d18328353R84) caused this code to require `.to_vec()` now.

**Comment by rafal-ch** [comment]
> When I run the node and make the following query:
> 
> ```
> curl -X POST http://localhost:4000/v1/graphql \
> -H "Content-Type: application/json" \
> -d '{
>     "query": "{ consensusParameters(version:0) { version blockGasLimit blockTransactionSizeLimit } }"
> }'
> ```
> 
> I get the following result
> 
> ```json
> {
>   "data": {
>     "consensusParameters": {
>       "version": "V2",
>       "blockGasLimit": "30000000",
>       "blockTransactionSizeLimit": "129024"
>     }
>   }
> }
> ```
> 
> Is this the expected result? I would imagine that by querying `consensusParameters(version:0)` we got still the V1 parameters, and to get the V2 consensus parameters we would need to run a query with `consensusParameters(version:1)`.
> 
> Or am I missing something?

My explanation would be that since the chain starts-up at genesis with [this change](https://github.com/FuelLabs/fuel-core/pull/2188/files#diff-a12c714e4dbb6f375ee2aef053e8d36ba844e1baf2439e07f407657cd96e3904R4) in the chainspec config included, it treats this as "version 0", aka "the first (zero-indexed) and only version known".

I assume that if we had started with the original chainspec containing `V1`, this `V1` would become a version `0`. And later if we upgrade the network to `V2`, this `V2` would actually become version `1`.

I not super certain this is the expected/correct behavior, though.

**Comment by acerone85** [comment]
Thanks for explaining. Overall LGTM.  I'll just wait for somebody to confirm that the behaviour re the version is correct, and that the breaking changes to fuel-vm are merged. After that I will be happy to approve.

File: CHANGELOG.md
Comment (AurelienFT) [inline CHANGELOG.md:18]: This description shouldn't be in this PR.

File: crates/client/assets/schema.sdl
Comment (AurelienFT) [inline crates/client/assets/schema.sdl:201]: When I first read it, I understood that it was the maximum size for a transaction. I think I misunderstood because the word transaction is singular. IMO, it should be plural.

**Review by AurelienFT — COMMENTED** [review]
Looks good to me, however I don't feel confortable approving this as I barely understand why each changes has been done.

**Comment by rafal-ch** [comment]
> Looks good to me, however I don't feel confortable approving this as I barely understand why each changes has been done.

It'll get more clear after https://github.com/FuelLabs/fuel-core/pull/2199 is merged.

**Review by xgreenx — APPROVED** [review]

**Review by xgreenx — APPROVED** [review]

File: crates/client/assets/schema.sdl
Comment (rafal-ch) [reply crates/client/assets/schema.sdl:201]: Yes, the name could indeed be slightly improved. However, it's already merged in `fuel-vm` [here](https://github.com/FuelLabs/fuel-vm/pull/821/files). 

In the follow-up PR to `fuel-core` I'll make sure that this is comprehensively documented.

File: CHANGELOG.md
Comment (rafal-ch) [reply CHANGELOG.md]: True, it's not necessarily a part of the feature being implemented in this PR, but the breaking changes was pulled into this branch [here](https://github.com/FuelLabs/fuel-core/pull/2188/commits/b7d8da650d9c8effe55d76b4d95b742ffb11da09) and [here](https://github.com/FuelLabs/fuel-core/pull/2188/commits/dba42fc76b496ddc02b72492cdbe3b61c8e966c6), so technically, this message belongs here.

I'll most likely not merge this PR before [the one which contained the original change](https://github.com/FuelLabs/fuel-core/pull/2199) is merged first.

**Review by Dentosal — APPROVED** [review]

**Comment by Dentosal** [comment]
Sorry about the accidental merge. Feel free to revert if needed.

**Pull request merged**
